### PR TITLE
Organize stakeholder's current frontend into our folder structure

### PR DESCRIPTION
**Pull request by yirano**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -20,12 +23,18 @@ import { ExampleDataViz } from './components/pages/ExampleDataViz';
 import { config } from './utils/oktaConfig';
 import { LoadingComponent } from './components/common';
```
Comment (Jahteo) [inline src/index.js]: Uneducated question: why the blank line here? Sometimes they make sense to me intuitively, sometimes I gotta ask to learn a new thing. Is it bc this is an import from one of our project files, that's not a component?

File: src/state/actions/index.js [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
 // This allows for the simplification of flow when importing actions into your components throughout your app.
-// Actions should be focused to a single purpose. 
-// You can have multiple action creators per file if it makes sense to the purpose those action creators are serving. 
-// Declare action TYPES at the top of the file
\ No newline at end of file
+// Actions should be focused to a single purpose.
+// You can have multiple action creators per file if it makes sense to the purpose those action creators are serving.
+// Declare action TYPES at the top of the file
+import axios from 'axios';
+
```
Comment (Jahteo) [inline src/state/actions/index.js:7]: I honestly don't know the answer to this: Is there a combineActions hook that we should be using like the combineReducers you did so well? (I hadn't even used combineReducer yet in a project, so I get to learn all the good things this week!)

**Review by Jahteo — COMMENTED** [review]
I like it. My Redux is rustier than I thought, so it's pointless for me to approve this personally, so I asked a couple minor informational questions for my own sake, and the rest seemed right. Love how organized you were with everything & how you broke it into several commits.

File: src/index.js [hunk 3]
```diff
@@ -20,12 +23,18 @@ import { ExampleDataViz } from './components/pages/ExampleDataViz';
...
 
+import reducer from './state/reducers/index';
+
+const store = createStore(reducer, applyMiddleware(thunk));
+
 ReactDOM.render(
-  <Router>
```
Comment (jidelson) [inline src/index.js:24]: I like how you separated this into another component.

File: package.json [hunk 4]
```diff
@@ -11,12 +11,19 @@
     "antd": "^4.4.3",
     "axios": "^0.19.2",
     "craco-less": "^1.17.0",
+    "moment": "^2.29.1",
+    "moment-timezone": "^0.5.32",
```
Comment (Jahteo) [inline package.json:15]: what's moment? why was this added?

File: src/index.js (hunk 1 above)
Comment (yirano) [reply src/index.js]: I group things by type -- libraries and components and such

File: src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react';
+import { BrowserRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import * as actions from '../state/actions/auth';
+
+import Landing from './pages/Landing/Landing';
+import Routes from './Routes';
+import '../style.css';
+
+class App extends Component {
```
Comment (jidelson) [inline src/components/App.js:10]: You mentioned you want to refactor this to a functional component. I am looking forward to that.

File: src/__tests__/FormInput.test.js [hunk 6]
```diff
@@ -23,3 +23,10 @@ describe('<FormInput /> test suite', () => {
     expect(input.placeholder).toBe('User Email');
   });
 });
+
+describe('Says Hello', () => {
```
Comment (Jahteo) [inline src/__tests__/FormInput.test.js:27]: You wrote a test to start! Yay!!

File: src/components/pages/Booking/Booking.js [hunk 7]
```diff
@@ -0,0 +1,399 @@
+import React from 'react';
+import axios from 'axios';
+import moment from 'moment';
+import 'moment-timezone';
+import { connect } from 'react-redux';
+import menteeComputer from '../../../assets/images/vbb-mentee-computer.png';
+
+class Booking extends React.Component {
```
Comment (jidelson) [inline src/components/pages/Booking/Booking.js:8]: Are you planning on refactoring this as well to a functional component?

File: src/components/AlertDismissable.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import Alert from 'react-bootstrap/Alert';
+
```
Comment (Jahteo) [inline src/components/AlertDismissable.js:3]: Do we want to use all the components that VBB started with or start from scratch?

File: src/state/actions/index.js (hunk 2 above)
Comment (yirano) [reply src/state/actions/index.js:7]: That's a really good question! I'm going to do some research.

File: src/components/App.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  componentDidMount() {
+    this.props.onTryAutoSignup();
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <BrowserRouter {...this.props}>
+          <Landing {...this.props}>
+            <Routes />
```
Comment (Jahteo) [inline src/components/App.js:20]: I'm torn. I like routes going elsewhere so that this file isn't too big, but I kinda want it here too. (not helpful)

File: src/components/App.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+          </Landing>
+        </BrowserRouter>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => {
+  return {
+    isAuthenticated: state.token !== null,
```
Comment (Jahteo) [inline src/components/App.js:30]: this is all the state that was used in VBB files, right?

File: src/__tests__/ExampleListContainer.test.js [hunk 11]
```diff
@@ -1,21 +1,21 @@
-import React from 'react';
-import { render, act, cleanup, waitForElement } from '@testing-library/react';
+// import React from 'react';
```
Comment (jidelson) [inline src/__tests__/ExampleListContainer.test.js]: Make sure we remember to come back to these commented out code/tests later

File: src/components/Routes.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    <div>
+      <PrivateRoute exact path="/" component={Dashboard} />
+      <PrivateRoute exact path="/booking/" component={Booking} />
+      <Route exact path="/signin/" component={HomeSignIn} />
+      <Route exact path="/signup/" component={MasterForm} />
+      <Route
+        exact
+        path="/sessiondetails/:sessionid/"
+        component={SessionDetails}
+      />
```
Comment (Jahteo) [inline src/components/Routes.js:22]: The way we planned this originally, had a route for each role dashboard. Do we want to move the role-dependent parts into the dashboard component instead?

File: src/components/pages/Booking/Booking.js [hunk 13]
```diff
@@ -0,0 +1,399 @@
+import React from 'react';
```
Comment (Jahteo) [inline src/components/pages/Booking/Booking.js:1]: DO we know if this file is functional? I know we're not working on this yet, but is it reusable, or something that'll be 100% scrapped?

File: src/components/pages/Dashboard/Dashboard.js [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+          sessionslots: res.data,
+        });
+      })
+      .catch(err => {
+        console.log(err);
+        alert('There was an error in retrieving your mentoring sessions', err);
+      });
+  };
+
+  componentDidMount() {
```
Comment (Jahteo) [inline src/components/pages/Dashboard/Dashboard.js:31]: Class components. Does it make other people want to rewrite it too? Maybe later when someone's bored?

File: src/components/pages/Home/HomeContainer.js [hunk 15]
```diff
@@ -1,42 +1,44 @@
 import React, { useState, useEffect, useMemo } from 'react';
```
Comment (Jahteo) [inline src/components/pages/Home/HomeContainer.js:1]: Was this file from SPA sample??

File: src/components/pages/HomeSignIn/HomeSignIn.js [hunk 16]
```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { GoogleLogin } from 'react-google-login';
+import vbbInAction from '../../../assets/images/vbb-in-action.png';
+import * as actions from '../../../state/actions/auth';
+
+export class HomeSignIn extends Component {
```
Comment (Jahteo) [inline src/components/pages/HomeSignIn/HomeSignIn.js:7]: Later, we should rename these to Login. later.

File: src/components/pages/Login/README.md [hunk 17]
```diff
@@ -1,7 +0,0 @@
-## Login and Authentication w/ Okta.
```
Comment (Jahteo) [inline src/components/pages/Login/README.md:1]: I kinda want this file, purely as a reference of what we should consider for google auth. for comparative purposes.

File: src/components/pages/Mentor/MentorProfile.js [hunk 18]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+function MentorProfile() {
+  return (
+    <div>
+      {false && ( //FIXME: get mentor profile information to display here
```
Comment (Jahteo) [inline src/components/pages/Mentor/MentorProfile.js:6]: This makes me happy.

File: src/components/pages/NotFound/NotFoundPage.js [hunk 19]
```diff
@@ -1,15 +0,0 @@
-import React from 'react';
```
Comment (Jahteo) [inline src/components/pages/NotFound/NotFoundPage.js:1]: I think we should keep a Not Found page component to build out later.

File: src/components/pages/Booking/Booking.js (hunk 7 above)
Comment (yirano) [reply src/components/pages/Booking/Booking.js:8]: I think we should! I'll get to doing that as part of my task -- adding it to Trello now.

File: src/components/pages/Registration/CardSection.js [hunk 20]
```diff
@@ -0,0 +1,35 @@
+/**
```
Comment (Jahteo) [inline src/components/pages/Registration/CardSection.js:1]: What does this do?

File: src/__tests__/ExampleListContainer.test.js (hunk 11 above)
Comment (yirano) [reply src/__tests__/ExampleListContainer.test.js]: We've got to or we're going to sit there wondering why life sucks. 😆

File: src/components/pages/Registration/MasterForm.js [hunk 21]
```diff
@@ -0,0 +1,259 @@
+import React from 'react';
```
Comment (Jahteo) [inline src/components/pages/Registration/MasterForm.js:1]: I'm glad this isn't a portion of our release.

File: src/state/actions/auth.js [hunk 22]
```diff
@@ -0,0 +1,155 @@
+import axios from 'axios';
```
Comment (Jahteo) [inline src/state/actions/auth.js:1]: I'll assume this works, eventually, I'd like to understand it. If/when we get to working on auth

File: src/state/reducers/index.js [hunk 23]
```diff
@@ -1,4 +1,9 @@
-// import all of your reducers into this file, and export them back out. 
+// import all of your reducers into this file, and export them back out.
 // This allows for the simplification of flow when importing reducers into your actions throughout your app.
 
+import { combineReducers } from 'redux';
+import headmasterReducer from './headmasterReducer.js';
 
+export default combineReducers({
```
Comment (Jahteo) [inline src/state/reducers/index.js:7]: I'd never actually used combineReducers, so I'm glad you set this up and I get to familiarize myself.

File: src/utils/PrivateRoute.js [hunk 24]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import Spinner from 'react-bootstrap/Spinner';
```
Comment (Jahteo) [inline src/utils/PrivateRoute.js:4]: ---> Ant design later.

**Review by Jahteo — APPROVED** [review]
I like it. Glad we got the ok from Frank to use their material. Amazed by how much you got done already and that you managed to find VBB's repo and incorporate it!!!

File: src/components/App.js (hunk 9 above)
Comment (yirano) [reply src/components/App.js:20]: This was actually what the Stakeholder did and I copied that over. I've never seen it done before either but I kind of like it.

**Review by jidelson — COMMENTED** [review]
Your practice request for review was successful!

File: package.json (hunk 4 above)
Comment (yirano) [reply package.json:15]: I think this is a date library that the stakeholder is using in one of their components.

**Review by jidelson — APPROVED** [review]
Take two. Should choose "Approve" and not "Comment"

File: src/components/pages/Registration/CardSection.js (hunk 20 above)
Comment (yirano) [reply src/components/pages/Registration/CardSection.js:1]: I think this is part of the stakeholder's "Donate" page
